src/main/kotlin/com/cultureamp/eventsourcing/EventTypeResolver.kt

```diff
@@ -0,0 +1,38 @@
+package com.cultureamp.eventsourcing
+
+import com.cultureamp.common.asNestedSealedConcreteClasses
+import kotlin.reflect.KClass
+
+interface EventTypeResolver {
+    fun serialize(domainEventClass: Class<out DomainEvent>): EventTypeDescription
+    fun deserialize(eventTypeDescription: EventTypeDescription): Class<out DomainEvent>
```

Comment:
This is a breaking change, right? WDYT about making an "upgrade notes.md" to say "when you upgrade to v0.25.0 you need to ..."

Reply:
Fair call - I just pushed up a commit with an entry in CHANGELOG.md following the precedent there for documenting breaking changes